src/test/java/pl/damianszczepanik/jenkins/buildhistorymanager/model/actions/DeleteArtifactsMatchingPatternsActionTest.java

```diff
@@ -0,0 +1,388 @@
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+import org.junit.runner.RunWith;
+import org.mockito.Mockito;
+import org.mockito.junit.MockitoJUnitRunner;
+
+import hudson.model.Run;
+import jenkins.model.ArtifactManager;
+import jenkins.util.VirtualFile;
+
+@RunWith(MockitoJUnitRunner.class)
+public class DeleteArtifactsMatchingPatternsActionTest {
+    private DeleteArtifactsMatchingPatternsAction action;
+    private DeleteArtifactsMatchingPatternsAction.DeleteFileCallable deleteFileCallableInstance;
+    private File archiveRootDirectory;
+    private Run<?, ?> mockRun;
+    private ArtifactManager mockArtifactManager;
+
+    @BeforeClass
+    public static void beforeClass() {
+        System.setProperty("java.util.logging.config.file",
+                ClassLoader.getSystemResource("logging.properties").getPath());
+    }
+
+    @Rule
+    public TemporaryFolder tempFolder = TemporaryFolder.builder().assureDeletion().build();
+
+    @Before
+    public void setup() throws IOException {
+        action = new DeleteArtifactsMatchingPatternsAction();
+        mockRun = Mockito.mock(Run.class);
+        mockArtifactManager = Mockito.mock(ArtifactManager.class);
+        Mockito.when(mockRun.getArtifactManager()).thenReturn(mockArtifactManager);
+        VirtualFile root = VirtualFile.forFile(tempFolder.getRoot());
+        Mockito.when(mockArtifactManager.root()).thenReturn(root);
+
+        // Create the tempFolder and its contents
+        tempFolder.newFile("test.xml");
+        tempFolder.newFile("testLog.log");
+        tempFolder.newFile("testTxt.txt");
+        tempFolder.newFolder("testFolder");
+        tempFolder.newFile("testFolder/test1.xml");
+        tempFolder.newFile("testFolder/testLog1.log");
+        tempFolder.newFile("testFolder/testTxt1.txt");
+
+        // Create a temporary directory for code coverage testing
+        Path tempDir = Files.createTempDirectory("testDir");
+        archiveRootDirectory = tempDir.resolve("archive").toFile();
+        //Set up DeleteFileCallable instance with the archiveRootDirectory
+        deleteFileCallableInstance = new DeleteArtifactsMatchingPatternsAction.DeleteFileCallable(archiveRootDirectory);
+    }
+
+    private void assertTempFolderContains(final int expectedNumFiles) {
+        final int actualNumFiles = countFiles(tempFolder.getRoot());
+        final String assertMessage = String.format("Expected %d files, actual found %d files", expectedNumFiles,
+                actualNumFiles);
+        Assert.assertEquals(assertMessage, expectedNumFiles, actualNumFiles);
+    }
+
+    // Custom assertFileExists method
+    public void assertFileExists(final String filePath, final String message) {
+        final File file = new File(tempFolder.getRoot(), filePath);
+        Assert.assertTrue(message, file.exists());
+    }
+
+    public void assertFileExists(final String filePath) {
```

Comment:
Can you rename method to more descriptive names e.g. https://github.com/jenkinsci/build-history-manager-plugin/blob/master/src/test/java/pl/damianszczepanik/jenkins/buildhistorymanager/model/conditions/BuildAgeRangeConditionTest.java#L17